web/src/main/java/org/cbioportal/web/parameter/StudyPageSettings.java

```diff
@@ -0,0 +1,256 @@
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+
+/**
+ * @author kalletlak
+ *
+ */
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(Include.NON_NULL)
+public class StudyPageSettings extends PageSettingsData implements Serializable {
+    
+    /**
+     * 
+     */
+    private static final long serialVersionUID = 1L;
+    
+    @JsonInclude(Include.NON_NULL)
+    static class ChartSetting implements Serializable {
+
+        /**
+         * 
+         */
+        private static final long serialVersionUID = 1L;
+
+        static class Layout implements Serializable {
+
+            /**
+             * 
+             */
+            private static final long serialVersionUID = 1L;
+
+            private Integer x;
+            private Integer y;
+            private Integer w;
+            private Integer h;
+
+            public Integer getX() {
+                return x;
+            }
+
+            public void setX(Integer x) {
+                this.x = x;
+            }
+
+            public Integer getY() {
+                return y;
+            }
+
+            public void setY(Integer y) {
+                this.y = y;
+            }
+
+            public Integer getW() {
+                return w;
+            }
+
+            public void setW(Integer w) {
+                this.w = w;
+            }
+
+            public Integer getH() {
+                return h;
+            }
+
+            public void setH(Integer h) {
+                this.h = h;
+            }
+        }
+
+        @JsonInclude(Include.NON_NULL)
+        static class PatientSampleIdentifier extends SampleIdentifier implements Serializable {
+            /**
+             * 
+             */
+            private static final long serialVersionUID = 1L;
+            private String patientId;
+
+            public String getPatientId() {
+                return patientId;
+            }
+
+            public void setPatientId(String patientId) {
+                this.patientId = patientId;
+            }
+
+            @Override
+            public boolean equals(Object o) {
+                if (o == this)
+                    return true;
+                if (!(o instanceof PatientSampleIdentifier)) {
+                    return false;
+                }
+                PatientSampleIdentifier patientSampleIdentifier = (PatientSampleIdentifier) o;
+                return Objects.equals(getSampleId(), patientSampleIdentifier.getSampleId())
+                        && Objects.equals(getPatientId(), patientSampleIdentifier.getPatientId())
+                        && Objects.equals(getStudyId(), patientSampleIdentifier.getStudyId());
+            }
+
+            @Override
+            public int hashCode() {
+                return Objects.hash(getSampleId(), getPatientId(), getStudyId());
+            }
+        }
+
+        @JsonInclude(Include.NON_NULL)
+        static class Group implements Serializable {
```

Comment:
Personally, `Group` is now more referring the group created for group comparison. Should we reconsider another name for the custom charts? Or call it `customChartGroup`?